resolwe/flow/executors/startup_processing_container.py

```diff
@@ -186,40 +188,69 @@ def process_script(self, message):
         try:
             yield from self._script_future
             self.return_code = self._script_future.result()
-        except asyncio.CancelledError:
-            pass
-
         finally:
             self._script_future = None
-            # Send final version of the log files.
-            logger.debug("Uploading latest log files.")
-            yield from self.manager.upload_log_files()
-
-            if KEEP_DATA:
+            try:
+                logger.debug("Uploading latest log files.")
                 try:
-                    yield from self.manager.send_file_descriptors(
-                        [str(file) for file in Path("./").rglob("*") if file.is_file()]
-                    )
+                    yield from self.manager.upload_log_files()
                 except:
-                    logger.exception("Error sending file descriptors (keep_data).")
-
-            response = respond(message, "OK", self.return_code)
-            yield from self.communicator.send_data(response)
-            # Close the cliet socket to notify communicator that processing is
-            # finished.
-            logger.debug("Closing upload socket.")
-            self.manager.upload_socket.close()
-            logger.debug("Upload socket closed.")
-
-            if self._script_finishing:
-                self._script_finishing.set()
-
-    def terminate_script(self):
-        """Terminate the running script."""
-        if self._script_future is not None:
-            logger.debug("Terminating script")
-            self._script_finishing = asyncio.Event()
-            self._script_future.cancel()
+                    yield from self._send_log(
+                        {"error": "Error uploading final log files."}
+                    )
+                    logger.exception("Error uploading final log files.")
+                    if self.return_code == 0:
+                        self.return_code = -1
+
+                if KEEP_DATA:
+                    try:
+                        yield from self.manager.send_file_descriptors(
+                            [
```

Comment:
Is there any advantage of using list comprehension here? If not, it would be shorter and cleaner to write a normal for loop.

Reply:
A for loop would require defining another variable and appending to it, making code longer.

I prefer list comprehension to for loop anytime: but if you still believe regular for loop is nicer I will rewrite the code.